### PR TITLE
First stab at adding global SmartStore

### PR DESCRIPTION
**Pull request by bhariharan**
TODO Items:

* Change the filter for deletion to exclude global DBs.
* Add new tests.
* Verify existing tests pass.

### CODE REVIEW

File: libs/SmartStore/src/com/salesforce/androidsdk/smartstore/app/SalesforceSDKManagerWithSmartStore.java [hunk 1]
```diff
@@ -218,6 +223,25 @@ public synchronized void changePasscode(String oldPass, String newPass) {
     }
 
     /**
+     * Returns the database used by smart store in the global context.
+     *
+     * @param dbName The database name. This must be a valid file name without a
+     * 				 filename extension such as ".db". Pass 'null' for default.
+     * @return SmartStore instance.
+     */
+    public SmartStore getGlobalSmartStore(String dbName) {
```
Comment (bhariharan) [inline libs/SmartStore/src/com/salesforce/androidsdk/smartstore/app/SalesforceSDKManagerWithSmartStore.java]: Passing in ```null``` or empty string for ```dbName``` here will use the default name. We could expose another method that takes no parameters if we want to, but there are already 4 or 5 of these types of methods here.

File: libs/SmartStore/src/com/salesforce/androidsdk/smartstore/app/SalesforceSDKManagerWithSmartStore.java [hunk 2]
```diff
@@ -309,4 +347,55 @@ public boolean hasSmartStore(UserAccount account, String communityId) {
...
      	return DBOpenHelper.smartStoreExists(context, dbNamePrefix, account, communityId);
      }
+
+     /**
+      * Removes the global smart store.
+      *
+      * @param dbName Database name. This must be a valid file name without a
+      * 			  filename extension such as ".db". Pass 'null' for default.
+      */
+     public void removeGlobalSmartStore(String dbName) {
```
Comment (bhariharan) [inline libs/SmartStore/src/com/salesforce/androidsdk/smartstore/app/SalesforceSDKManagerWithSmartStore.java]: ```remove``` was never exposed here before, although it was possible through ```DbOpenHelper```.

**Comment by bhariharan** [comment]
Added re-encryption of global DBs in [this](https://github.com/bhariharan/SalesforceMobileSDK-Android/commit/59b6449e2ef383fcafc1afff09dd43f74e329107) commit.

File: libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/DBOpenHelper.java [hunk 3]
```diff
@@ -95,20 +105,19 @@ public static synchronized DBOpenHelper getOpenHelper(Context ctx,
 	 * @param communityId Community ID.
 	 * @return DBOpenHelper instance.
 	 */
-	public static DBOpenHelper getOpenHelper(Context ctx, String dbNamePrefix, UserAccount account, String communityId) {
-		StringBuffer dbName = new StringBuffer(dbNamePrefix);
-		
-		// If we have account information, we will use it to create a database suffix for the user
+	public static DBOpenHelper getOpenHelper(Context ctx, String dbNamePrefix,
```
Comment (khawkins) [inline libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/DBOpenHelper.java]: Ah, so like you said, this functionality already existed.

**Comment by khawkins** [comment]
Looks good to me, over all.

**Comment by wmathurin** [comment]
Looks fine.

**Comment by bhariharan** [comment]
Looks like we already had tests for global smart store in ```DbOpenHelperTest```. Nothing for me to do there :-)

**Comment by bhariharan** [comment]
See [this](https://github.com/bhariharan/SalesforceMobileSDK-Android/commit/8dda2173f43c8d41137b6b9bf175223315fe2222) commit for the last round of changes.

* Added tests for removal of DBs.
* Fixed the file filter logic, so that global DBs are not removed upon logout.
* Misc. cleanup.

**Comment by khawkins** [comment]
:+1:

**Pull request merged**